runtime/packages/core-modules/assembly/system.ts

```diff
@@ -90,9 +95,97 @@ export class System {
      */
     static incAccountNonce(who: AccountId): void{
         const oldNonce = System.accountNonce(who);
-        const nonceKey: u8[] = Utils.stringsToU8a([System.NONCE_KEY]);
+        const nonceKey: u8[] = Utils.stringsToBytes([System.NONCE_KEY], true);
         const newNonce = new UInt64(oldNonce.value + 1);
         Storage.set(who.getAddress().concat(nonceKey), newNonce.toU8a());
     }
+
+    /**
+    * Maximum number of block number to block hash mappings to keep (oldest pruned first).
+    */
+    static blockHashCount(): UInt32 {
+        const value = Storage.get(Utils.stringsToBytes(["system", "block_cout"], true));
+        if(value.isSome()){
+            const decValue = Bytes.decodeCompactInt((<ByteArray>value.unwrap()).values);
+            return new UInt32(<u32>decValue.value);
+        }
+        return new UInt32(0);
+    }
+
+    /**
+     * Gets the index of extrinsic that is currently executing.
+     */
+    static extrinsicIndex(): UInt32{
+        const extIndex = Storage.take(Utils.stringsToBytes([System.EXTRINSIC_INDEX], true));
+        return UInt32.fromU8a(extIndex);
+    }
+
+    /**
+     * Increment extrinsic index
+     */
+    static incExtrinsicIndex(): void {
+        const count = this.extrinsicIndex();
+        const newCount = new UInt32(count.value + 1);
+        Storage.set(Utils.stringsToBytes([System.EXTRINSIC_INDEX], true), newCount.toU8a());
+    }
+
+    /**
+     * Computes the extrinsicsRoot for the given data and populates storage
+     * @param data 
+     */
+    static deriveExtrinsics(): void{
```

Comment:
`derive` sounds like you are going to return something. I think that a more suitable name would be `computeExtrinsicsRoot`